app/models/user.rb

```diff
@@ -80,15 +80,13 @@ def self.create_with_omniauth(auth, league = nil)
   end
 
   def current_streak
-    wins = []
-    matches.each do |m|
-      if m.win_for?(self)
-        wins << m
-      else
-        break
-      end
+    scope = Match.won_by(id)
+    if date_of_last_lost_match = Match.lost_by(id).select(:date).reorder('date DESC').first
+      scope = scope.where('date > ?', date_of_last_lost_match.date)
     end
-    wins.length
+    scope.count
+  rescue NoMethodError => ex
```

Comment:
Remove it move it move it. Remove it move it move it. 

![tumblr_m4wnnxgadp1r8dy8go1_500](https://cloud.githubusercontent.com/assets/573120/10846429/086a1c78-7f10-11e5-9ba6-3432156719a4.gif)